### PR TITLE
install_mask: remove masked symlinks to directories (bug 678462)

### PR DESCRIPTION
**Pull request by thesamesam**
os.walk() categorizes symlinks to directories as directories so they
were being ignored by INSTALL_MASK. This change calls os.scandir()
instead which efficiently provides more control.

[sam: cherry-picked from chromiumos' third_party/portage_tool repo]
(cherry picked from commit 6473079d07351dda49a906cb89d24a9a39526bf7)
Bug: https://bugs.gentoo.org/678462
Signed-off-by: Sam James <sam@gentoo.org>

### CODE REVIEW

File: lib/portage/tests/util/test_install_mask.py [hunk 1]
```diff
@@ -166,3 +166,29 @@ def testTrailingSlash(self):
...
+            install_mask = InstallMask("/foo/")
+            install_mask_dir(tmp_dir, install_mask)
+            self.assertFalse(
+                os.path.lexists(link_name), "failed to remove {}".format(link_name)
+            )
+            self.assertFalse(
+                os.path.lexists(base_dir), "failed to remove {}".format(base_dir)
+            )
+        finally:
+            shutil.rmtree(tmp_dir)
```
Comment (zmedico) [inline lib/portage/tests/util/test_install_mask.py:194]: Looks like this file is missing imports for tempfile, shutil, and the install_mask_dir function.

Comment (thesamesam) [reply lib/portage/tests/util/test_install_mask.py:194]: sorry, yes! Had to apply manually due to black. I realised but internet cut out, fixing now

**Review by zmedico — APPROVED** [review]
LGTM

**Comment by thesamesam** [comment]
cheers again zac!

**Pull request closed without merging**